application/models/LSActiveRecord.php

```diff
@@ -424,7 +424,7 @@ public static function decryptSingleOld($value = ''): string
     public static function encryptSingle($value = '')
     {
         // if $value is provided, it would decrypt
-        if (!empty($value)) {
+        if (isset($value) && $value !== "") {
```

Comment:
If value is "", shouldn't that be encrypted too? Isn't that a piece of data to be encrypted?

I believe that we have 2 problems:
1 - How to encrypt empty data (0, null, "", .. ) everything that matches empty()
2 - On this case I would just add an else.
```
 if (!empty($value)) {
 } else {
   return $value;
}
```

Reply:
before we use empty, the broke with 0 … it's the issue …

Then you 2 cause the issue 18501 … _Answers with 0 as value was not saved if encrypted_

Reply:
> empty strings should be encrypted as well

This is true.

Reply:
Another issue to report, right ?

Reply:
It's a bit complicated as numerical columns _cannot_ be encrypted (nowhere to save the hash), but of course people can write "0" in text questions. :)

Reply:
> It's a bit complicated as numerical columns cannot be encrypted (nowhere to save the hash), but of course people can write "0" in text questions. :)

If you have a numerical question encrypted : it was saved in a text column … 

All crypted columns are text columns … date or numérical or single choice.

Reply:
> Another issue to report, right ?

Depends on if this fix breaks it or not :sweat_smile:

Reply:
It's clearly another issue than «Answers with 0 as value was not saved if encrypted»

And more 4 month after …

Reply:
> > It's a bit complicated as numerical columns cannot be encrypted (nowhere to save the hash), but of course people can write "0" in text questions. :)
> 
> If you have a numerical question encrypted : it was saved in a text column …
> 
> All crypted columns are text columns … date or numérical or single choice.

Oh OK, didn't know.